packages/amesos2/src/Amesos2_TpetraMultiVecAdapter_def.hpp

```diff
@@ -211,23 +212,14 @@ namespace Amesos2 {
           const size_t lclNumRows = redist_mv.getLocalLength();
           for (size_t j = 0; j < redist_mv.getNumVectors(); ++j) {
             auto av_j = av(lda*j, lclNumRows);
-            auto X_j = redist_mv.getVector(j);
             auto X_lcl_j_2d = redist_mv.template getLocalView<host_execution_space> ();
             auto X_lcl_j_1d = Kokkos::subview (X_lcl_j_2d, Kokkos::ALL (), j);
-            for ( size_t i = 0; i < lclNumRows; ++i ) {
-              av_j[i] = X_lcl_j_1d(i);
-            }
+
+            using val_type = typename decltype( X_lcl_j_1d )::value_type;
+            Kokkos::View<val_type*, Kokkos::HostSpace> umavj ( const_cast< val_type* > ( reinterpret_cast<const val_type*> ( av_j.getRawPtr () ) ), av_j.size () );
```

Comment:
How come `av_j` is const?  Now is not the time to dig into that, though ;-) .

Reply:
The RHS MV (i.e. 'B') is stored as const in the MVAdapter (after being copied on root proc) to prevent modification, which is why `av_j` ends up being `const`.